### PR TITLE
Bump egui

### PR DESCRIPTION
**Pull request by okiwi6**
## Introduced Changes

This PR bumps the egui version to 0.24.
It also removes the fork of egui-dock from @knoellle since his changes got merged.
Upgrading egui-dock to the newest egui version also brought some fancy new features into twix.

Fixes #

## ToDo / Known Issues

*If this is a WIP describe which problems are to be fixed.*

## Ideas for Next Iterations (Not This PR)

*If there are some improvements that could be done in a next iteration, describe them here.*

## How to Test

* make sure twix still works

### CODE REVIEW

File: tools/twix/src/main.rs [hunk 1]
```diff
@@ -148,7 +151,7 @@ struct TwixApp {
     ip_address: String,
     panel_selection: String,
     last_focused_tab: (NodeIndex, TabIndex),
-    tree: Tree<Tab>,
+    tree: DockState<Tab>,
```
Comment (knoellle) [inline tools/twix/src/main.rs:154]: rename member to `dock(_state)`?

File: tools/twix/src/panels/image/mod.rs [hunk 2]
```diff
@@ -165,17 +165,15 @@ impl ImagePanel {
...
-        let image_rect = Rect::from_min_size(image_response.rect.left_top(), displayed_image_size);
-        let painter = TwixPainter::paint_at(ui, image_rect).with_camera(
+        let image_identifier = format!("bytes://image-{:?}", self.cycler_selector);
+        ui.ctx().forget_image(&image_identifier);
+        let image = Image::from_bytes(image_identifier, image_raw)
+            .texture_options(TextureOptions::NEAREST)
+            .fit_to_fraction(Vec2::splat(1.0));
+        let image_response = ui.add(image);
+        // let displayed_image_size = image_size * scale;
+        // let image_rect = Rect::from_min_size(image_response.rect.left_top(), displayed_image_size);
```
Comment (knoellle) [inline tools/twix/src/panels/image/mod.rs:175]: Remove these comments

**Review by knoellle — CHANGES_REQUESTED** [review]
Looks good for the most part, I left some minor remarks.

File: tools/twix/src/main.rs (hunk 1 above)
Comment (okiwi6) [reply tools/twix/src/main.rs]: done

File: tools/twix/src/panels/image/mod.rs (hunk 2 above)
Comment (okiwi6) [reply tools/twix/src/panels/image/mod.rs]: done

**Review by knoellle — APPROVED** [review]
LGTM

**Pull request merged**